topics/statistics/tutorials/RNN/tutorial.md

```diff
@@ -0,0 +1,364 @@
+---
+layout: tutorial_hands_on
+
+title: Introduction to recurrent neural networks (RNN)
+zenodo_link: https://zenodo.org/record/4477881
+questions:
+- What is a recurrent neural network (RNN)?
+- What are some applications of RNN?
+objectives:
+- Understand the difference between feedforward neural networks (FNN) and RNN
+- Learn various RNN types and architectures
+- Solve a sentiment analysis problem on IMDB movie review dataset using RNN in Galaxy
+requirements:
+  -
+    type: internal
+    topic_name: statistics
+    tutorials:
+      - intro_deep_learning
+time_estimation: 2H
+contributors:
+- kxk302
+
+---
+
+
+
+
+# Introduction
+{:.no_toc}
+
+<!-- This is a comment. -->
+
+Artificial neural networks are a machine learning discipline roughly inspired by how neurons in a 
+human brain work. In the past decade, there has been a huge resurgence of neural networks thanks 
+to the vast availability of training data and enormous increases in computing capacity (Successfully 
+training complex neural networks in some domains requires lots of data and compute capacity). There 
+are various types of neural networks (Feedforward, recurrent, etc). In this tutorial, we discuss 
+the recurrent neural networks and explain how they differ from feedforward variants. We also describe 
+various RNN architectures and solve a sentiment analysis problem using RNN in Galaxy.       
+
+> ### Agenda
+>
+> In this tutorial, we will cover:
+>
+> 1. TOC
+> {:toc}
+>
+{: .agenda}
+
+# Review of feedforward neural networks (FNN)
+
+In feedforward neural networks (FNN) all the fields of a training example are presented
+to the network at once, after which the the network generates an output. For example, a
+lung X-ray image is passed to a FNN, and the network predicts tumor or no tumor. By contrast,
+in RNN the training example fields are presented to the network one at a time. For example,
+a sequence of English words is passed to an RNN, one at a time, and the network generates a
+sequence of Persian words, one at a time. RNN handle sequential data, whether its temporal or ordinal.
+
+## Single layer FNN
+
+![Alternative text](../../images/FFNN_no_hidden.png "Single layer feedforward neural network")
+
+Figure 1 shows a single layer FNN, where the input is 3 dimensional. Each input field is multiplied by a
+weight. Afterwards, the results are summed up, along with a bias, and passed to an activation function.
+
+![Alternative text](../../images/activation.gif "Activation of the output neuron o1. Activation function f could be Sigmoid, Tanh, ReLU, etc.")
+
+The activation function can have many forms (sigmoid, tanh, ReLU, linear, step function, sign function, etc.).
```

Comment:
It would be nice to have links to these activation functions such as sigmoid, tanh and so on for further reading.

Reply:
Added a reference to a paper that reviews activation functions.